src/Spice86.Core/Emulator/InterruptHandlers/Dos/Ems/ExpandedMemoryManager.cs

```diff
@@ -0,0 +1,877 @@
+﻿using Serilog;
+using Serilog.Events;
+
+using Spice86.Logging;
+
+using System.Numerics;
+
+namespace Spice86.Core.Emulator.InterruptHandlers.Dos.Ems;
+
+using Spice86.Core.Emulator.Callback;
+using Spice86.Core.Emulator.Errors;
+using Spice86.Core.Emulator.InterruptHandlers;
+using Spice86.Core.Emulator.Memory;
+using Spice86.Core.Emulator.OperatingSystem;
+using Spice86.Core.Emulator.VM;
+
+using System.Linq;
+
+/// <summary>
+/// Provides DOS applications with EMS memory.
+/// </summary>
+public sealed class ExpandedMemoryManager : InterruptHandler {
+    /// <summary>
+    /// Size of each EMS page in bytes.
+    /// </summary>
+    public const int PageSize = 16384;
+
+    /// <summary>
+    /// Maximum number of mappable pages.
+    /// </summary>
+    public const int MaximumPhysicalPages = 4;
+
+    /// <summary>
+    /// Maximum number of logical pages.
+    /// </summary>
+    public const int MaximumLogicalPages = 256;
+
+    public const ushort PageFrameSegment = 0xE000;
+    public const int FirstHandle = 1;
+    public const int LastHandle = 254;
+    public const int SystemHandle = 0;
+
+    public const string EmsIdentifier = "EMMXXXX0";
+
+    public Memory ExpandedMemory { get; init; } = new(32 * 1024);
+
+    public override ushort? InterruptHandlerSegment => 0xF100;
+
+    private readonly short[] _pageOwners = new short[MaximumLogicalPages];
+    private readonly SortedList<int, EmsHandle> _handles = new();
+    private readonly int[] _mappedPages = new int[] {-1, -1, -1, -1};
+
+    private readonly ILoggerService _loggerService;
+    
+    public ExpandedMemoryManager(Machine machine, ILoggerService loggerService) : base(machine) {
+        _loggerService = loggerService;
+        MemoryUtils.SetZeroTerminatedString(ExpandedMemory.Ram, MemoryUtils.ToPhysicalAddress(0xF100 - PageFrameSegment, 0x000A), EmsIdentifier, EmsIdentifier.Length + 1);
```

Comment:
Not sure I understand why this is setting a string in the EMS card

Reply:
Yup, this doesn't make sense. Fixed !